homeassistant/components/switcher_kis/switch.py

```diff
@@ -0,0 +1,294 @@
+        """Return the device's ip address."""
+        ret = self._device_data.ip_addr  # type: str
+        return ret
+
+    @property
+    def electric_current(self) -> Optional[float]:
+        """Return the electric current."""
+        ret = self._device_data.electric_current  # type: Optional[float]
+        return ret
+
+    @property
+    def remaining_time(self) -> Optional[str]:
+        """Return the remaining time to off command."""
+        ret = self._device_data.remaining_time  # type: Optional[str]
+        return ret
+
+    @property
+    def auto_off_set(self) -> str:
+        """Return the auto off configuration set."""
+        ret = self._device_data.auto_off_set  # type: str
+        return ret
+
+    @property
+    def last_data_update(self) -> datetime:
+        """Return the datetime for the last update received by the device."""
+        ret = self._device_data.last_data_update  # type: datetime
+        return ret
+
+    @property
+    def last_state_change(self) -> datetime:
+        """Return the datetime for the last state change."""
+        ret = self._device_data.last_state_change  # type: datetime
+        return ret
+
+    @property
+    def device_name(self) -> str:
+        """Return the device's name."""
+        ret = self._device_data.name  # type: str
+        return ret
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True if entity has to be polled for state.
+
+        False if entity pushes its state to HA.
+        """
+        return False
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return "{}-{}".format(
+            self._device_data.device_id, self._device_data.mac_addr)
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def state(self) -> str:
+        """Return the state of the entity."""
+        return (
+            STATE_ON if self._state == SWITCHER_STATE_ON
+            else STATE_OFF if self._state == SWITCHER_STATE_OFF
+            else STATE_UNKNOWN)
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if entity is on."""
+        return self.state == STATE_ON
+
+    @property
+    def current_power_w(self) -> Optional[int]:
+        """Return the current power usage in W."""
+        ret = self._device_data.power_consumption  # type: Optional[int]
+        return ret
+
+    @property
+    def state_attributes(self) -> Dict:
+        """Return the optional state attributes."""
+        attribs = {}
+
+        for prop, attr in PROPERTIES_TO_ATTRIBUTES.items():
+            value = getattr(self, prop)
+            if value:
+                attribs[attr] = value
+
+        return attribs
+
+    @property
+    def icon(self) -> Optional[str]:
+        """Return the icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def available(self) -> bool:
```

Comment:
`STATE_UNKNOWN` and `STATE_UNAVAILABLE` are different things.  Do not use one to determine the other.